org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/themes/Theme.java

```diff
@@ -1,158 +1,179 @@
+	 * 
+	 * @param element
+	 */
+	public Theme(String id, String path, String name, boolean dark, boolean isDefault) {
+		super(path);
+		this.id = id;
+		this.name = name;
+		this.dark = dark;
+		this.isDefault = isDefault;
+	}
+
+	public Theme(IConfigurationElement ce) {
+		super(ce);
+		id = ce.getAttribute(XMLConstants.ID_ATTR);
+		name = ce.getAttribute(XMLConstants.NAME_ATTR);
+		dark = "true".equals(ce.getAttribute(DARK_ATTR));
+		isDefault = "true".equals(ce.getAttribute(DEFAULT_ATTR));
+	}
+
+	@Override
+	public String getId() {
+		return id;
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public IToken getToken(String type) {
+		ITokenProvider provider = getTokenProvider();
+		return provider != null ? provider.getToken(type) : null;
+	}
+
+	@Override
+	public Color getEditorForeground() {
+		ITokenProvider provider = getTokenProvider();
+		Color themeColor = provider != null ? provider.getEditorForeground() : null;
+		return ColorManager.getInstance()
+				.getPriorityColor(themeColor, AbstractTextEditor.PREFERENCE_COLOR_FOREGROUND);
+	}
+
+	@Override
+	public Color getEditorBackground() {
+		ITokenProvider provider = getTokenProvider();
+		Color themeColor = provider != null ? provider.getEditorBackground() : null;
+		return ColorManager.getInstance()
+				.getPriorityColor(themeColor, AbstractTextEditor.PREFERENCE_COLOR_BACKGROUND);
+	}
+
+	@Override
+	public Color getEditorSelectionForeground() {
+		ITokenProvider provider = getTokenProvider();
+		Color themeColor = provider != null ? provider.getEditorSelectionForeground() : null;
+		return ColorManager.getInstance()
+				.getPriorityColor(themeColor, AbstractTextEditor.PREFERENCE_COLOR_SELECTION_FOREGROUND);
+	}
+
+	@Override
+	public Color getEditorSelectionBackground() {
+		ITokenProvider provider = getTokenProvider();
+		Color themeColor = provider != null ? provider.getEditorSelectionBackground() : null;
+		return ColorManager.getInstance()
+				.getPriorityColor(themeColor, AbstractTextEditor.PREFERENCE_COLOR_SELECTION_BACKGROUND);
+	}
+
+	@Override
+	public Color getEditorCurrentLineHighlight() {
+		ITokenProvider provider = getTokenProvider();
+		Color themeColor = provider != null ? provider.getEditorCurrentLineHighlight() : null;
+		ColorManager manager = ColorManager.getInstance();
+		return manager.isColorUserDefined(AbstractTextEditor.PREFERENCE_COLOR_BACKGROUND)
+				? manager.getPreferenceEditorColor(PreferenceConstants.EDITOR_CURRENTLINE_HIGHLIGHT)
+				: themeColor;
+	}
+
+	private ITokenProvider getTokenProvider() {
+		if (tokenProvider == null) {
+			try {
+				InputStream in = super.getInputStream();
+				if (in == null) {
+					return null;
+				}
+				tokenProvider = new CSSTokenProvider(in);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		return tokenProvider;
+	}
+
+	@Override
+	public String toCSSStyleSheet() {
+		return super.getResourceContent();
+	}
+
+	@Override
+	public boolean isDark() {
```

Comment:
Would be good to find something.more dynamic, like some.`match(themeId)` or `match(Color backgroundColor)` to allow more cases.

Reply:
I agree with you, but we should change it later, not in this PR.